common/json_tok.c

```diff
@@ -584,3 +584,76 @@ struct command_result *param_extra_tlvs(struct command *cmd, const char *name,
 	*fields = temp;
 	return NULL;
 }
+
+struct command_result *param_routehint(struct command *cmd, const char *name,
+				       const char *buffer, const jsmntok_t *tok,
+				       struct route_info **ri)
+{
+	size_t i;
+	const jsmntok_t *curr;
+	char *element_name;
+	const char *err;
+
+	if (tok->type != JSMN_ARRAY) {
+		return command_fail(
+		    cmd, JSONRPC2_INVALID_PARAMS,
+		    "Routehint %s (\"%s\") is not an array of hop objects",
+		    name, json_strdup(tmpctx, buffer, tok));
+	}
+
+	*ri = tal_arr(cmd, struct route_info, 0);
+	json_for_each_arr(i, curr, tok) {
+		struct route_info e;
+		struct amount_msat temp;
+		element_name = tal_fmt(cmd, "%s[%zu]", name, i);
```

Comment:
Creating this var here seems weird?  It's only used once, and inside a format string?

Reply:
Copy pasta from the outer loop, will remove :+1: